src/game_picture.cpp

```diff
@@ -204,30 +204,30 @@ void Game_Picture::Update() {
 	if (data.time_left == 0) {
 		SyncCurrentToFinish();
 	} else {
-		auto interpolate = [=](double& current, double finish) {
+		auto interpolate = [=](double current, double finish) {
 			double d = data.time_left;
-			current = (current * (d - 1) + finish) / d;
+			return (current * (d - 1) + finish) / d;
```

Comment:
Because before _interpolate_ was overwriting _current_ with the new value instead of returning it.